pkg/backend/httpstate/token_source_test.go

```diff
@@ -84,6 +89,94 @@ func TestTokenSourceWithQuicklyExpiringInitialToken(t *testing.T) {
 	}
 }
 
+func TestTokenSourceWithClient(t *testing.T) {
+	t.Parallel()
+
+	response := "timeout"
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if response == "timeout" {
+			w.WriteHeader(http.StatusGatewayTimeout)
+			_, err := w.Write([]byte("Gateway Timeout"))
+			require.NoError(t, err)
+		} else {
+			w.WriteHeader(http.StatusForbidden)
+			_, err := w.Write([]byte("Forbidden"))
+			require.NoError(t, err)
+		}
+	}))
+	defer server.Close()
+
+	apiClient := client.NewClient(server.URL, "fake-token", true, nil)
+	ctx := context.Background()
+
+	clock := clockwork.NewFakeClock()
+
+	tok0, tok0Expires := "fake-token", clock.Now().Add(20*time.Millisecond)
+
+	duration := 20 * time.Millisecond
+	assumedExpires := func() time.Time {
+		return time.Now().Add(duration)
+	}
+
+	updateIdentifier := client.UpdateIdentifier{
+		StackIdentifier: client.StackIdentifier{
+			Owner:   "my-owner",
+			Project: "my-project",
+			Stack:   tokens.MustParseStackName("my-stack"),
+		},
+		UpdateKind: apitype.PreviewUpdate,
+		UpdateID:   "my-update",
+	}
+
+	renewLeaseCalled := 0
+	renewLease := func(
```

Comment:
Is it possible to use the original `renewLease` function here, and wrap it to increment `renewLeaseCalled`? Otherwise we might break the actual function, but not updated the test.

Reply:
Good idea.  I made it so now.